File: ruby/lib/task_master/output_handlers/task_distribution_report/terminal_handler.rb
module TaskMaster
  module OutputHandlers
    module TaskDistributionReport
      class TerminalHandler
        def generate_output(data)
          AsciiTable.new(headers, DataProcessor.new(data).process).draw
        end

        private

        def headers
          ["Assignee", "Count"]
        end

        class DataProcessor
          def initialize(data)
            @data = data
          end

          def process
            @data.map(&:assignee).uniq.sort.map do |assignee_name|
              [assignee_name, @data.count { |x| x.assigned_to?(assignee_name) }.to_s]
            end
          end
        end
      end
    end
  end
end

File: javascript/lib/outputHandlers/taskDistributionReport/TerminalHandler.js
const path = require('path');
const AsciiTable = require(path.join(__dirname, '../../AsciiTable'));

class TerminalHandler {
	constructor() {
		this.AsciiTable = AsciiTable
		this.DataProcessor = class {
			constructor(data) {
				this.data = data
			}

			process() {
				return [...new Set(this.data.map(item => item.assignee))].sort().map(assigneeName => [
					assigneeName,
					this.data.filter(item => item.assignee === assigneeName).length.toString()
				])
			}
		}
	}

	generateOutput(data) {
		return new this.AsciiTable(this.#headers(), new this.DataProcessor(data).process()).draw()
	}

	#headers() {
		return ["Assignee", "Count"]
	}
}

module.exports = TerminalHandler;

